pkg/scheduler/partition.go

```diff
@@ -61,6 +59,7 @@ type PartitionContext struct {
 	userGroupCache         *security.UserGroupCache        // user cache per partition
 	totalPartitionResource *resources.Resource             // Total node resources
 	nodeSortingPolicy      *policies.NodeSortingPolicy     // Global Node Sorting Policies
+	allocations            int                             // Number of allocations on the partition
```

Comment:
why we still need to track the number of allocations?
if this is just for UT, I think we should remove them.

Reply:
No the UI uses this and shows this as number when it gets the partition overview.

Reply:
Is it better to re-implement `GetTotalAllocationCount()`, and simply loop all the apps and sum allocations of all apps?
So we don't need to maintain a separate `allocations` count here.

Reply:
That would require us to grab a copy of the apps (i.e. lock the partition). Then we need to lock each app when we get the allocation count from the app. Really heavy and not scalable as soon as we go above a small number of apps and allocations per app.

Reply:
I don't think that's gonna be a problem for perf. It only needs the app's read lock to get a count. And this is only called via REST API. But if we do that, the code will be simpler, and fewer objects to maintain (worry about keeping them inconsistent all the time). And note, update the count here is not a lock-free call, it holds an even bigger partition lock.

Reply:
Created https://issues.apache.org/jira/browse/YUNIKORN-526 to follow up on this.